Explorer/Assets/Scripts/ECS/StreamableLoading/GLTF/LoadGLTFSystem.cs

```diff
@@ -0,0 +1,121 @@
+            gltfDownloadProvider = new GltFastDownloadProvider(World, sceneData, partitionComponent);
+        }
+
+        // Might be used later
+        // private AnimationMethod GetAnimationMethod()
+        // {
+        //     string sPlatform = PlatformUtils.GetPlatform();
+        //     BuildTarget bt = BuildTarget.StandaloneWindows64; // default
+        //
+        //     switch (sPlatform)
+        //     {
+        //         case "_windows":
+        //         {
+        //             bt = BuildTarget.StandaloneWindows64;
+        //             break;
+        //         }
+        //         case "_mac":
+        //         {
+        //             bt = BuildTarget.StandaloneOSX;
+        //             break;
+        //         }
+        //         case "_linux":
+        //         {
+        //             bt = BuildTarget.StandaloneLinux64;
+        //             break;
+        //         }
+        //     }
+        //
+        //     return bt is BuildTarget.StandaloneWindows64 or BuildTarget.StandaloneOSX
+        //         ? AnimationMethod.Mecanim
+        //         : AnimationMethod.Legacy;
+        // }
+
+        protected override async UniTask<StreamableLoadingResult<GLTFData>> FlowInternalAsync(GetGLTFIntention intention, IAcquiredBudget acquiredBudget, IPartitionComponent partition, CancellationToken ct)
+        {
+            if (!sceneData.SceneContent.TryGetContentUrl(intention.Name!, out var finalDownloadUrl))
+                return new StreamableLoadingResult<GLTFData>(new Exception("The content to download couldn't be found"));
+
+            gltfDownloadProvider.TargetGltfOriginalPath = intention.Name!;
+            var gltfImport = new GltfImport(downloadProvider: gltfDownloadProvider, logger: gltfConsoleLogger);
+
+            var gltFastSettings = new ImportSettings
+            {
+                NodeNameMethod = NameImportMethod.OriginalUnique,
+                AnisotropicFilterLevel = 0,
+                GenerateMipMaps = false,
+            };
+
+            bool success = await gltfImport.Load(finalDownloadUrl, gltFastSettings, ct);
+
+            // Release budget now to not hold it until dependencies are resolved to prevent a deadlock
+            acquiredBudget.Release();
+
+            if (success)
+            {
+                // We do the GameObject instantiation in this system since 'InstantiateMainSceneAsync()' is async.
+                var rootContainer = new GameObject(gltfImport.GetSceneName(0));
+
+                // Let the upper layer decide what to do with the root
+                rootContainer.SetActive(false);
+
+                await InstantiateGltfAsync(gltfImport, rootContainer.transform);
+
+                return new StreamableLoadingResult<GLTFData>(new GLTFData(gltfImport, rootContainer));
+            }
+
+            return new StreamableLoadingResult<GLTFData>(new Exception("The content to download couldn't be found"));
+        }
```

Comment:
Although it's a good suggestion to use pooling whenever possible, in this case it could introduce more complexity than it's worth since these `GameObject` can't easily be re-used and returning them to the pool could be more compute intense.